src/core/lombok/core/handlers/HandlerUtil.java

```diff
@@ -980,4 +989,13 @@ public static String getParamJavadoc(String methodComment, String param) {
 		}
 		return null;
 	}
+
+	/**
+	 * Determines the list of non-null annotations that should be used.
+	 * @param useForeignAnnotations the config parameter, if we want to use foreign non-null annotations (default: true).
+	 * @return depending on the config, it either returns a list of all known non-null annotations or just containing the lombok.NonNull annotation
+	 */
+	public static List<String> nonNullAnnotations(Boolean useForeignAnnotations) {
+		return Boolean.FALSE.equals(useForeignAnnotations) ? LOMBOK_NONNULL_ANNOTATIONS : NONNULL_ANNOTATIONS;
```

Comment:
Here is the switch logic, which non-null annotations should lombok use. By default, it uses all known nonNull-annotations. if `lombok.nonNull.useForeignAnnotations` is set to false, only the lombok.NonNull annotations (there is currently only one, but we are prepared, if an experimental one is added)

So people, who do not want that lombok use foreign annotations can turn it off now.

It would be also possible to specify a list of nonNull annotations in the configuration. What do you think?